### ISSUE TITLE
Switch to React.js to guarantee the plugin UI integrity

### ISSUE THREAD

**Issue by Natim** [mozilla/notes#537]
Today states are handled using multiple global variables, it is really brittle in order to keep having a sustainable development we want to switch to React/Redux.

**Vendored libraries**

- jose-jwe-jws → jose.js
- fxa-crypto-relier → fxa-crypto-relier.js
- kinto-http → kinto-http.js
- kinto → kinto.js
- testpilot-ga → testpilot-ga.js
- material-design-lite → material.js
- quill → quill.js

**Files number of lines**

```
$ find . -name '*.js' | egrep -v vendor | xargs wc -l
   54 ./settings/settings.js
  356 ./sync.js
  288 ./sidebar/panel.js
   87 ./sidebar/migration.js
   32 ./sidebar/theme.js
  164 ./sidebar/editor.js
   87 ./fxa-utils.js
  175 ./background.js
 1243 total
```

**Comment by Natim** [comment]
We might want to use this in our webpack conf in order to build both the background script and the panel scripts with one webpack config: https://github.com/webpack/webpack/issues/1189#issuecomment-156576084

**Comment by sebastienbarbier** [comment]
Also using webpack for build ad web-ext for rendering might create some conflict on browser refresh. Issue is discussed on web-ext repo: https://github.com/mozilla/web-ext/issues/1102

**Comment by Natim** [comment]
Fixed with #550
